### PR TITLE
Code cleanup for unused __arrayTranslateTRTOSimpleVMX function

### PR DESCRIPTION
**Pull request by mnalam-p**
Apparently this function was introduced to compress strings on Z arch.
However, this function is not used any more and is being treated as
dead code for 2y+.

Signed-off-by: Mohammad Nazmul Alam <mohammad.nazmul.alam@ibm.com>

### CODE REVIEW

File: runtime/compiler/p/runtime/J9PPCArrayTranslate.spp [hunk 1]
```diff
@@ -1313,973 +1264,9 @@ __arrayTranslateTROT:
...
-
-!----------------------------------------------------------------------
-! macro definitions
-!    macroArrayTranslate swHeader,swSrcDstAlign,swBlock4Loop,swBlock2,swBlock1,swTrailer
-!      macroArrayTranslateHeader
-!      macroArrayTranslateUnrollInit
-!      macroArrayTranslateUnrolledBodyN swSrcDstAlign num
-!        macroArrayTranslateUnrolledBodyGen vin0 vin2 swSrcDstAlign outReg firstBlock
-!      macroArrayTranslateTrailer
-!----------------------------------------------------------------------
```
Comment (IBMJimmyk) [inline runtime/compiler/p/runtime/J9PPCArrayTranslate.spp]: From here downward it looks like you are also getting rid of a bunch of macro definitions that appear to be unrelated to `arrayTranslateTRTOSimpleVMX`. They are behind an `#if 0` so I'm not sure if they are really needed anymore. But you should check to make sure.

File: runtime/compiler/p/runtime/J9PPCArrayTranslate.spp [hunk 2]
```diff
@@ -1313,973 +1264,9 @@ __arrayTranslateTROT:
...
-L.vector111110:
-	macroArrayTranslate(`1',`1',`1',`1',`1',`0')
-	.align	5
-L.vector111111:
-	macroArrayTranslate(`1',`1',`1',`1',`1',`1')
-
-!----------------------------------------------------------------------
-! endof Handlers
-!----------------------------------------------------------------------
-    	endproc.__arrayTranslateTRTOSimpleVMX:
```
Comment (IBMJimmyk) [inline runtime/compiler/p/runtime/J9PPCArrayTranslate.spp]: Just wanted to highlight that this line does need to be removed even if you aren't removing the above macros. This line should actually be with that rest of the `arrayTranslateTRTOSimpleVMX` but probably got moved apart due to a merge error and was never caught because it was behind an `#if 0`.

**Comment by IBMJimmyk** [comment]
It looks like `runtime/compiler/runtime/Runtime.cpp` also has references to `arrayTranslateTRTOSimpleVMX` that can be removed.

**Comment by fjeremic** [comment]
Hi @mnalam-p, congrats on your first contribution. Your commit/PR message could use some more detail. Please see our contribution guidelines for what is expected at the eclipse/openj9 project:
https://github.com/eclipse/openj9/blob/master/CONTRIBUTING.md#commit-guidelines

> The first line describes the change made. It is written in the imperative mood, and should say what happens when the patch is applied. Keep it short and simple. The first line should be less than 70 characters, where reasonable, and should be written in sentence case preferably not ending in a period. Leave a blank line between the first line and the message body.
>
> The body should be wrapped at 72 characters, where reasonable.
>
> Include as much information in your commit as possible. You may want to include designs and rationale, examples and code, or issues and next steps. Prefer copying resources into the body of the commit over providing external links. Structure large commit messages with headers, references etc. Remember, however, that the commit message is always going to be rendered in plain text.
> ...
> ...

**Comment by gita-omr** [comment]
@zl-wang maybe you have some background on that code and whether it can be removed?

**Comment by zl-wang** [comment]
don't remember where it came from originally.  likely Tokyo folks worked that into our code base. it looked not performing to me.  i would support removing it.

However, you have to remove all related code though, including OMR side.  There are two files in openj9 side, 3 files in OMR side.

File: runtime/compiler/p/runtime/J9PPCArrayTranslate.spp (hunk 1 above)
Comment (mnalam-p) [reply runtime/compiler/p/runtime/J9PPCArrayTranslate.spp]: @IBMJimmyk thanks for the feedback. My thought was, because they were not used for 2 yo+, maybe they are good candidate for cleanup. Also, because they were defined within the .startproc and .endproc.

**Comment by fjeremic** [comment]
Thanks @mnalam-p, I've edited the PR description to match your commit comment. One last thing to fix up is the sentence case of the commit and PR and a slight typo:

```
Code cleanup for unused __arrayTranslateTRTOSimpleVMX function

Apparently this function was introduced to compress strings on Z arch.
However, this function is not used any more and is being treated as
dead code for 2y+.

Signed-off-by: Mohammad Nazmul Alam <mohammad.nazmul.alam@ibm.com>
```

**Comment by mnalam-p** [comment]
@fjeremic updated commit message. Thank you for your feedback.

**Comment by mnalam-p** [comment]
related PR: https://github.com/eclipse/omr/pull/5000
to cleanup the usage of the function in OMR

**Comment by fjeremic** [comment]
Is this change a hard or soft dependency on the OMR change? That is, will merging the OMR change break the OpenJ9 build?

**Comment by mnalam-p** [comment]
I tested this PR only with official OMR, it passed. Hence, I can only speculate that it is a soft dependency.

**Comment by mnalam-p** [comment]
@IBMJimmyk pr updated to reflect your suggested changes

**Comment by fjeremic** [comment]
Jenkins test sanity plinux,aix jdk8
